paymentsheet/src/main/java/com/stripe/android/paymentsheet/paymentdatacollection/polling/PollingActivity.kt

```diff
@@ -14,39 +25,64 @@ internal class PollingActivity : AppCompatActivity() {
         requireNotNull(PollingContract.Args.fromIntent(intent))
     }
 
-    private val listener = FragmentResultListener { _, result ->
-        handleResult(result)
+    private val viewModel by viewModels<PollingViewModel> {
+        PollingViewModel.Factory {
+            PollingViewModel.Args(
+                clientSecret = args.clientSecret,
+                timeLimit = args.timeLimitInSeconds.seconds,
+                initialDelay = args.initialDelayInSeconds.seconds,
+                maxAttempts = args.maxAttempts,
+            )
+        }
     }
 
+    @OptIn(ExperimentalMaterialApi::class)
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        supportFragmentManager.setFragmentResultListener(
-            PollingFragment.KEY_FRAGMENT_RESULT,
-            this,
-            listener
-        )
+        WindowCompat.setDecorFitsSystemWindows(window, false)
 
-        args.statusBarColor?.let { color ->
-            window.statusBarColor = color
-        }
+        setContent {
+            StripeTheme {
+                val state = rememberBottomSheetState(confirmValueChange = { false })
```

Comment:
We don’t want users to be able to dismiss the sheet by swiping, so `confirmValueChange` returns `false`.